src/leap/mail/imap/tests/test_imap.py

```diff
@@ -982,89 +1079,106 @@ def select():
         def check():
             return self.client.check()
 
-        d = self.connected.addCallback(strip(login))
+        d = self.connected.addCallbacks(
+            strip(add_mailbox), self._ebGeneral)
+        d.addCallbacks(lambda _: login(), self._ebGeneral)
         d.addCallbacks(strip(select), self._ebGeneral)
         d.addCallbacks(strip(check), self._ebGeneral)
         d.addCallbacks(self._cbStopClient, self._ebGeneral)
-        return self.loopback()
-
-        # Okay, that was fun
-
-    def testClose(self):
-        """
-        Test closing the mailbox. We expect to get deleted all messages flagged
-        as such.
-        """
-        name = 'mailbox-close'
-        self.server.theAccount.addMailbox(name)
-
-        m = LeapIMAPServer.theAccount.getMailbox(name)
-
-        def login():
-            return self.client.login(TEST_USER, TEST_PASSWD)
-
-        def select():
-            return self.client.select(name)
-
-        def add_messages():
-            d1 = m.messages.add_msg(
-                'test 1', subject="Message 1",
-                flags=('\\Deleted', 'AnotherFlag'))
-            d2 = m.messages.add_msg(
-                'test 2', subject="Message 2",
-                flags=('AnotherFlag',))
-            d3 = m.messages.add_msg(
-                'test 3', subject="Message 3",
-                flags=('\\Deleted',))
-            d = defer.gatherResults([d1, d2, d3])
-            return d
-
-        def close():
-            return self.client.close()
-
-        d = self.connected.addCallback(strip(login))
-        d.addCallbacks(strip(select), self._ebGeneral)
-        d.addCallbacks(strip(add_messages), self._ebGeneral)
-        d.addCallbacks(strip(close), self._ebGeneral)
-        d.addCallbacks(self._cbStopClient, self._ebGeneral)
         d2 = self.loopback()
-        return defer.gatherResults([d, d2]).addCallback(self._cbTestClose, m)
-
-    def _cbTestClose(self, ignored, m):
-        self.assertEqual(len(m.messages), 1)
-        msg = m.messages.get_msg_by_uid(2)
-        self.assertTrue(msg is not None)
-
-        self.assertEqual(
-            dict(msg.hdoc.content)['subject'],
-            'Message 2')
-        self.failUnless(m.closed)
+        return defer.gatherResults([d, d2])
+
+        # Okay, that was much fun indeed
+
+    # skipping close test: we just need expunge for now.
```

Comment:
Something that I find a bit cleaner to skip a test is to rename it to `_testClose`. I found that done in keymanager and I like it. But anyway, not really important to fix.